app/models/customer.rb

```diff
@@ -1,2 +1,12 @@
 class Customer < ApplicationRecord
+  has_one :enrollment, dependent: :restrict_with_error
+
+  def hire_plan!(plan)
+    if enrollment
+      enrollment.plan = plan
+      enrollment.save!
+    else
+      Enrollment.create(customer: self, plan: plan, status: :active)
```

Comment:
Como active é o padrão, aqui podemoria simplificar essa linha com o método de criação do has_one do current_user. Será que funciona?

```suggestion
      create_enrollment(plan: plan)
```